plugins/Capacity.py

```diff
@@ -0,0 +1,147 @@
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+
+class Capacity(Plugin):
+    def init(self, logger):
+        Plugin.init(self, logger)
+
+        self.errors[30912] = self.def_class(
+            item=3091,
+            level=2,
+            tags=["tag"],
+            title=T_("Invalid capacity value"),
+            detail=T_("""Capacity tags should be positive integers."""),
+            resource="https://wiki.openstreetmap.org/wiki/Key:capacity",
+        )
+        self.errors[30913] = self.def_class(
+            item=3091,
+            level=2,
+            tags=["tag"],
+            title=T_("Specific capacity is greater than total capacity"),
+            detail=T_("""A capacity:* value is greater than the total capacity."""),
+            resource="https://wiki.openstreetmap.org/wiki/Key:capacity",
+        )
+
+    def node(self, data, tags):
+        errors = []
+        if ("capacity" not in tags
+                # Ignore errors that should be reported by generic analysers
+                or tags["capacity"] == ""):
+            return []
+        try:
+            total_capacity = int(tags["capacity"])
+            if total_capacity < 0:
+                return [{
+                    "class": 30913,
+                    "subclass": 5,
+                    "text": T_('"{0}" value "{1}" is negative', "capacity", total_capacity),
+                }]
+        except ValueError:
+            errors.append({
+                "class": 30912,
+                "subclass": 4,
+                "text": T_('"{0}" value "{1}" is not an integer', "capacity", tags["capacity"]),
+            })
+            total_capacity = None
+
+        for key, value in tags.items():
+            if (
+                not key.startswith("capacity:")
+                # Ignore errors that should be reported by generic prefix analysers
+                or key == "capacity:"
+                or value == ""
+
+                or value in ("yes", "no")
```

Comment:
```suggestion
                or value in ("", "yes", "no")
```

Reply:
I split those two checks as `""` is an invalid value that should be checked in other Analysers, while `yes|no` are valid values. I believe the performance penalty is really low for the clarity added by [the comment L71](https://github.com/osm-fr/osmose-backend/pull/1023/files#diff-d376487ccb6816192f500423b8279401ea86f138eb7be2bbb05c26567be043e9R71-R73).

Reply:
Ok. So just remove the empty line, please.

Reply:
I reordered the conditions to remove the empty line while still keeping the comment relevant, I hope that's okay.